packages/ia-components/sandbox/details/DetailsDownloadOptions.jsx

```diff
@@ -22,37 +25,40 @@ import { I18nSpan, I18nStr, I18nIcon } from "../languages/Languages";
 
 export default class DetailsDownloadOptions extends React.Component {
   downloadableFilesDict() {
-    return this.props.files.reduce((res, af) => {
-      const format = af.metadata.format;
-      if (downloadableFormat(format)) {
-        if (!res[format]) { res[format] = []; }
-        res[format].push(af);
-      }
-      return res;
-    }, {});
+    return this.props.files
+      .filter(af => ( !this.props.disconnected || af.downloaded))
+      .reduce((res, af) => {
+        const format = af.metadata.format;
+        if (downloadableFormat(format)) {
+          if (!res[format]) { res[format] = []; }
+          res[format].push(af);
+        }
+        return res;
+      }, {});
```

Comment:
a few items to note - 
- I don't see a constructor that explicitly binds this class state to the function yet this function is calling `this.props`.
- What happens when `files` is not an array? Will it bork immediately?

Reply:
I'm not sure the problem here, its a method on the class called from the render (to make render easier to ready), it doesn't need to be bound because as a class method it always has access to this (its only onClick etc which for some react-bizarre reason don't get passed "this") 

It should barf on files not being an array - that would be a coding error and AFAIK we dont expect code to do extreme checking for errors that would be coding errors (as opposed to data errors).